teuthology/test/test_proc_regex.py

```diff
@@ -0,0 +1,56 @@
+import argparse
+
+from teuthology.orchestra import cluster
+from teuthology import misc
+from teuthology.orchestra.daemon.group import DaemonGroup
+import subprocess
+
+
+class FakeRemote(object):
+    pass
+
+
+def test_get_daemon_pid():
+    ps_output = """
+    ceph       658     1  0 Jun08 ?        00:07:43 /usr/bin/ceph-mgr -f --cluster ceph --id host1 --setuser ceph --setgroup ceph
+    ceph      1634     1  0 Jun08 ?        00:02:17 /usr/bin/ceph-mds -f --cluster ceph --id host1 --setuser ceph --setgroup ceph
+    ceph     31555     1  0 Jun08 ?        01:13:50 /usr/bin/ceph-mon -f --cluster ceph --id host1 --setuser ceph --setgroup ceph
+    ceph     31765     1  0 Jun08 ?        00:48:42 /usr/bin/radosgw -f --cluster ceph --name client.rgw.host1.rgw0 --setuser ceph --setgroup ceph
+    ceph     97427     1  0 Jun17 ?        00:41:39 /usr/bin/ceph-osd -f --cluster ceph --id 0 --setuser ceph --setgroup ceph
+    """
+
+    ctx = argparse.Namespace()
+    ctx.daemons = DaemonGroup(use_systemd=True)
+    remote = FakeRemote()
+
+    def sh(args):
+        args[0] = "echo"
+        args[1] = '"{}"'.format(ps_output)
+        print(args[2:])
+        args = " ".join([arg.value if hasattr(arg, "value")
+                         else arg for arg in args])
+        return subprocess.getoutput(args)
```

Comment:
Instead of system calls use better to use Mock.

Reply:
Not using Mock here, because trying to test it with all daemons.